code/modules/holodeck/computer.dm

```diff
@@ -110,63 +120,193 @@
 	. = TRUE
 	switch(action)
 		if("load_program")
-			var/program_to_load = text2path(params["type"])
-			if(!ispath(program_to_load))
-				return FALSE
-			var/valid = FALSE
-			var/list/checked = program_cache.Copy()
-			if(obj_flags & EMAGGED)
-				checked |= emag_programs
-			for(var/prog in checked)
-				var/list/P = prog
-				if(P["type"] == program_to_load)
-					valid = TRUE
-					break
-			if(!valid)
-				return FALSE
-
-			var/area/A = locate(program_to_load) in GLOB.sortedAreas
-			if(A)
-				load_program(A)
+			var/program_to_load = params["id"]
+			//load the map_template that program_to_load represents
+			if(program_to_load)
+				load_program(program_to_load)
```

Comment:
You seem to have dropped the check for if the template actually exists and if emagged template is loaded into actually emagged holodeck.